src/components/Api.js

```diff
@@ -0,0 +1,65 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _getRequest(endPoint, method = "GET") {
+    return fetch(this._baseUrl + endPoint, {
+      method: method,
+      headers: this._headers,
+    })
+      .then(this._checkSuccessResponce)
+      .catch((err) => {
+        console.log(err);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.